h2o-algos/src/main/java/hex/glm/GLMUtils.java

```diff
@@ -281,4 +284,28 @@ public static double calSmoothNess(double[][] beta, double[][][] penaltyMatrix,
     }
     return smoothval;
   }
+
+  /***
+   * Estimate dispersion factor using maximum likelihood.  I followed section IV of the doc in 
+   * https://h2oai.atlassian.net/browse/PUBDEV-8683 . 
+   */
+  public static double estimateMLSE(GLMTask.ComputeMLSETsk mlCT, double seInit) {
+    double constantValue = mlCT._sumlnyi + mlCT._wsum - mlCT._sumlnui - mlCT._sumyiOverui;
+    while (true) {
```

Comment:
is it always ensured that this loop will end in finite time?

Reply:
From my tests, it does

Reply:
but i will add a progress check to stop if max runtimr is exceeded.